benchmarks/structured/iqpe/iqpe.qasm

```diff
@@ -0,0 +1,23 @@
+OPENQASM 3.0;
+include "qelib1.inc";
+
+input int precision;
+
+qubit q;
+qubit anc;
+bit[precision] res;
+
+x anc;
+
+for int i in [precision - 1:0:-1] {
+    h q;
+    pow(2**i) @ cp(3*pi/8) q, anc;
```

Comment:
```suggestion
    ctrl @ pow(2**i) @ p(3*pi/8) q, anc;
```
I think this would be even more idiomatic iQPE as the textbook algorithm applies controlled powers of a unitary.

Any reasonable compiler would simplify the above to
```qasm
cp((2**i)*3*pi/8) q, anc;
```
(or even further)